.github/workflows/build.yaml

```diff
@@ -0,0 +1,34 @@
+name: build and test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  POETRY_HOME: "/opt/poetry"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: |
+            3.9
+            3.10
+            3.11
```

Comment:
3.12 as well?

Reply:
Not sure pyarrow supports 3.12. It's not listed here https://arrow.apache.org/docs/python/install.html#python-compatibility

Reply:
yeah, I tried that then failed with earlier pyarrow releases. now I use py312 with pyarrow14, but don't test the earlier ones with that since pyarrow installation fails.